### DISCUSSION TITLE
Error: This page crashed. Invariant failed: You should not use <Switch> outside a <Router>

### DISCUSSION THREAD

**Discussion by cramatt** [facebook/docusaurus#6884] [Q&A]
![image](https://user-images.githubusercontent.com/22599518/157475764-84dac312-5ce8-498a-9f5c-6a3694398c31.png)

I have docusaurus installed in a yarn monorepo. I have tried the above Bug report "Prerequisites" but they are not realistic in a monorepo and cause more errors, for example: 

- Deleting yarn.lock for a workspace causes unexpected updates to many modules which results in a broken workspace.
- I'm don't believe its possible to create a monorepo reproduction in https://new.docusaurus.io/.
- `yarn clear` does not work for me (maybe I'm on an older version?)

Initially I was seeing yarn warnings like this: 
```
client (webpack 5.70.0) compiled successfully
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Resolving 'file-loader/dist/cjs' in users/site/node_modules/url-loader/dist for build dependencies doesn't lead to expected result 'users/site/node_modules/file-loader/dist/cjs.js', but to 'users/site/node_modules/url-loader/node_modules/file-loader/dist/cjs.js' instead. Resolving dependencies are ignored for this path.
<w>  at unknown 4 file-loader/dist/cjs
<w>  at file dependencies users/site/node_modules/url-loader/dist/index.js
<w>  at file users/site/node_modules/url-loader/dist/index.js
<w>  at file dependencies users/site/node_modules/url-loader/dist/cjs.js
<w>  at file users/site/node_modules/url-loader/dist/cjs.js
<w>  at resolve commonjs users/site/node_modules/url-loader/dist/cjs.js
```

So I updated my workspace package to this: 

```
"nohoist": [
    "@docusaurus/*"
  ],
```

And now the yarn warnings are gone. However the docs site is still broken. 

If I delete my node modules folders, and delete yarn.lock, the docs site DOES INSTALL AND RUN correctly, but then my other projects in the workspace are broken. 

Any thoughts on a fix?

**Comment by slorber** [comment]
Unfortunately it is very complicated to debug such kind of issue without a repro

I suggest regenerating all lockfiles, upgrading and reinstalling everything

And removing all your swizzled components as you might have a bug inside one of them

**Reply by cramatt** [reply]
Thanks @slorber for the reply. As mentioned above, if I delete my lockfile and reinstall things DO WORK. However this is not really a great "fix" since it breaks other packages in the workspace, since many dependencies have upgraded. 

Specifically I'm seeing an issue with maybe create-react-app updating, and the TS version getting more strict? (changes from 'react' to 'react-jsx'). 

Anyhow, I agree its difficult to debug with a repro, let me see if I can put something together...

**Reply by slorber** [reply]
Unfortunately, it's your monorepo setup, I can't really help to debug it as I don't even have access and it would be time-consuming. 

Try to nohoist everything, and progressively add back monorepo hoisting one package after another until you find the problem 😅

**Reply by cramatt** [reply]
No worries! I do have this currently

```
"nohoist": [
    "@docusaurus/*"
  ],
```

I think the issue is actually due to incorrect Type files for one of the other packages dependencies
